spec/shopify_transporter/exporters/magento/product_exporter_spec.rb

```diff
@@ -32,14 +33,69 @@ module Magento
 
             expected_result = [
               {
-                product_id: 12345,
+                product_id: '12345',
+                type: 'simple',
                 top_level_attribute: "an_attribute",
               },
             ]
 
-            exporter = described_class.new(store_id: 1, client: soap_client)
+            exporter = described_class.new(store_id: 1, soap_client: soap_client)
             expect(exporter.export).to eq(expected_result)
           end
+
+          it 'retrieves configurable products from Magento using the SOAP API and injects simple product ids' do
+            soap_client = double("soap client")
+            product_mapping_exporter = double("product_mapping_exporter")
+
+            catalog_product_list_response_body = double('catalog_product_list_response_body')
+            catalog_product_info_response_body = double('catalog_product_info_response_body')
+
+            expect(soap_client)
+              .to receive(:call).with(:catalog_product_list, anything)
+              .and_return(catalog_product_list_response_body)
+              .at_least(:once)
+
+            expect(ProductMappingExporter).to receive(:new).and_return(product_mapping_exporter)
+            expect(product_mapping_exporter).to receive(:write_mappings)
+
+            expect(catalog_product_list_response_body).to receive(:body).and_return(
+              catalog_product_list_response: {
+                store_view: {
+                  item: [
+                    {
+                      product_id: '12345',
+                      type: 'configurable',
+                      top_level_attribute: "an_attribute",
+                    },
+                  ],
+                },
+              },
+            ).at_least(:once)
+
+            expected_result = [
+              {
+                product_id: '12345',
+                top_level_attribute: "an_attribute",
+                type: 'configurable',
+                simple_product_ids: ['800', '801', '802'],
```

Comment:
I think we set a `parent_ids` instead, this will make the conversion step easier. It's easier to look up/work with configurable products than it is to look up simple products.

so this might be similar to:
{
  product_id: '800',
  ...
  parent_id: '12345'
}

Reply:
Done.